lib/internal/url.js

```diff
@@ -1500,38 +1500,85 @@ function fileURLToPath(path, options = kEmptyObject) {
   return (windows ?? isWindows) ? getPathFromURLWin32(path) : getPathFromURLPosix(path);
 }
 
-// The following characters are percent-encoded when converting from file path
-// to URL:
-// - %: The percent character is the only character not encoded by the
-//        `pathname` setter.
-// - \: Backslash is encoded on non-windows platforms since it's a valid
-//      character but the `pathname` setters replaces it by a forward slash.
-// - LF: The newline character is stripped out by the `pathname` setter.
-//       (See whatwg/url#419)
-// - CR: The carriage return character is also stripped out by the `pathname`
-//       setter.
-// - TAB: The tab character is also stripped out by the `pathname` setter.
+// RFC1738 defines the following chars as "unsafe" for URLs
+// @see https://www.ietf.org/rfc/rfc1738.txt 2.2. URL Character Encoding Issues
+// Back-slashes must be special-cased on Windows, where they are treated as path separator.
 const percentRegEx = /%/g;
 const backslashRegEx = /\\/g;
 const newlineRegEx = /\n/g;
 const carriageReturnRegEx = /\r/g;
 const tabRegEx = /\t/g;
-const questionRegex = /\?/g;
+const quoteRegEx = /"/g;
 const hashRegex = /#/g;
+const spaceRegEx = / /g;
+const lowerThanRegEx = /</g;
+const greaterThanRegEx = />/g;
+const questionRegex = /\?/g;
+const openSquareBracketRegEx = /\[/g;
+const closeSquareBracketRegEx = /]/g;
+const openCurlyBracketRegEx = /{/g;
+const closeCurlyBracketRegEx = /}/g;
+const caretRegEx = /\^/g;
+const backtickRegEx = /`/g;
+const verticalBarRegEx = /\|/g;
 
 function encodePathChars(filepath, options = kEmptyObject) {
-  const windows = options?.windows;
-  if (StringPrototypeIndexOf(filepath, '%') !== -1)
+  if (StringPrototypeIncludes(filepath, '%')) {
     filepath = RegExpPrototypeSymbolReplace(percentRegEx, filepath, '%25');
-  // In posix, backslash is a valid character in paths:
-  if (!(windows ?? isWindows) && StringPrototypeIndexOf(filepath, '\\') !== -1)
-    filepath = RegExpPrototypeSymbolReplace(backslashRegEx, filepath, '%5C');
-  if (StringPrototypeIndexOf(filepath, '\n') !== -1)
+  }
+
+  if (StringPrototypeIncludes(filepath, '\t')) {
```

Comment:
Just out of curiousity, what's faster:
`StringPrototypeIncludes`
`StringPrototypeReplace`
`RegexpPrototypeExec`
etc?

Reply:
According to the benchmarks in this PR, it's `StringPrototypeIncludes`+`RegExpPrototypeSymbolReplace`